scripts/breaking_changes_checker/tests/test_changelog.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import json
+import jsondiff
+from breaking_changes_checker.breaking_changes_tracker import BreakingChangesTracker, ChangeType
+
+
+def test_changelog_flag():
+    with open(os.path.join(os.path.dirname(__file__), "examples", "code-reports", "content-safety", "stable.json"), "r") as fd:
+        stable = json.load(fd)
+    with open(os.path.join(os.path.dirname(__file__), "examples", "code-reports", "content-safety", "current.json"), "r") as fd:
+        current = json.load(fd)
+    diff = jsondiff.diff(stable, current)
+
+    bc = BreakingChangesTracker(stable, current, diff, "azure-ai-contentsafety", changelog=True)
+    bc.run_checks()
+
```

Comment:
Do we not need to call bc.report_changelog() here?

Reply:
No, since I'm not checking against the formatted output. I'm just checking that we got the features reported. Here is the tracking issue for expanding tests to also check the output of the report_changelog method: https://github.com/Azure/azure-sdk-for-python/issues/36064